core-analysis/01-filter-sce.R

```diff
@@ -237,16 +244,21 @@ if (opt$filtering_method == "manual") {
   }
 }
 
-# Remove old gene-level rowData statistics and recalculate
-drop_cols = colnames(rowData(filtered_sce)) %in% c('mean', 'detected')
-rowData(filtered_sce) <- rowData(filtered_sce)[!drop_cols]
-filtered_sce <- scater::addPerFeatureQC(filtered_sce)
-
-# Filter the genes (rows)
-detected <-
-  rowData(filtered_sce)$detected > opt$gene_detected_row_cutoff
-expressed <- rowData(filtered_sce)$mean > opt$gene_means_cutoff
-filtered_sce <- filtered_sce[detected & expressed, ]
+if(opt$filter_genes) {
+  # Remove old gene-level rowData statistics and recalculate
+  drop_cols = colnames(rowData(filtered_sce)) %in% c('mean', 'detected')
+  rowData(filtered_sce) <- rowData(filtered_sce)[!drop_cols]
+  filtered_sce <- scater::addPerFeatureQC(filtered_sce)
+  
+  # Filter the genes (rows)
+  detected <-
+    rowData(filtered_sce)$detected > opt$gene_detected_row_cutoff
+  expressed <- rowData(filtered_sce)$mean > opt$gene_means_cutoff
+  filtered_sce <- filtered_sce[detected & expressed,]
+  metadata(filtered_sce)$genes_filtered <- "genes filtered"
+} else {
+  metadata(filtered_sce)$genes_filterd <- "no genes filtered"
```

Comment:
```suggestion
} 
metadata(filtered_sce)$genes_filtered <- opt$filter_genes
```
I think we can actually just do this and set it to true/false which will be the value of the argument so we can do it on one line.